src/metabase/upload/util.cljc

```diff
@@ -0,0 +1,58 @@
+(ns metabase.upload.util)
+
+(defn unique-name-generator
+  "A fork of [[mbql.u/unique-name-generator]] which accounts for collisions when names are truncated.
+  There is due to a limitation where it does not play well with truncation, see the tests for this function.
+  This may be a bit slower, and it's a breaking change as far as what ids look like, hence it's a fork."
```

Comment:
The breaking change is that we would no longer produce IDs like `blah_2_2`, rather it would become `blah_3`.